### PR TITLE
[alertmanager] Allow exposing configmap reloaders port

### PR DESCRIPTION
**Pull request by Sadzeih**
Needed for scraping reloader metrics

Signed-off-by: Alexis Guerville <alexis.guerville@messagebird.com>

#### What this PR does / why we need it:

This allows exposing ports on the configmapreloaders containers. It is necessary to allow scraping the reloaders metrics.

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/alertmanager/values.yaml [hunk 1]
```diff
@@ -172,6 +172,10 @@ configmapReload:
     tag: v0.5.0
     pullPolicy: IfNotPresent
 
+  ## container port
+  ##
+  # port: 9533
```
Comment (naseemkullah) [inline charts/alertmanager/values.yaml:177]:
```suggestion
  # containerPort: 9533
```

File: charts/alertmanager/templates/statefulset.yaml [hunk 2]
```diff
@@ -62,6 +62,10 @@ spec:
             - --webhook-url=http://127.0.0.1:{{ .Values.service.port }}/-/reload
           resources:
             {{- toYaml .Values.configmapReload.resources | nindent 12 }}
+          {{- if .Values.configmapReload.port }}
+          ports:
+            - containerPort: {{ .Values.configmapReload.port }}
```
Comment (naseemkullah) [inline charts/alertmanager/templates/statefulset.yaml:67]:
```suggestion
          {{- if .Values.configmapReload.containerPort }}
          ports:
            - containerPort: {{ .Values.configmapReload.containerPort }}
```

**Review by naseemkullah — DISMISSED** [review]
Thanks!

nit: port => containerPort removes need for comment in values file

File: charts/alertmanager/templates/statefulset.yaml [hunk 3]
```diff
@@ -62,6 +62,10 @@ spec:
             - --webhook-url=http://127.0.0.1:{{ .Values.service.port }}/-/reload
           resources:
             {{- toYaml .Values.configmapReload.resources | nindent 12 }}
+          {{- if .Values.configmapReload.containerPort }}
+          ports:
+            - containerPort: {{ .Values.configmapReload.containerPort }}
```
Comment (naseemkullah) [inline charts/alertmanager/templates/statefulset.yaml:67]: im not totally familiar with the reloader, is there any relation to the webhook-url port?

File: charts/alertmanager/templates/statefulset.yaml [hunk 4]
```diff
@@ -62,6 +62,10 @@ spec:
...
+          {{- end }}
```
Comment (naseemkullah) [inline charts/alertmanager/templates/statefulset.yaml:68]:
```suggestion
          ports:
            - containerPort: 9533
```

Would this be bad?

Comment (Sadzeih) [reply charts/alertmanager/templates/statefulset.yaml:68]: I didn't want my change to affect every user of the chart, that's why I kept it optional.

Comment (naseemkullah) [reply charts/alertmanager/templates/statefulset.yaml:68]: True, I appreciate that, but now I am thinking what if we did, what would be the worst case scenario?

Comment (Sadzeih) [reply charts/alertmanager/templates/statefulset.yaml:68]: People might already use the `9533` port number?

Comment (naseemkullah) [reply charts/alertmanager/templates/statefulset.yaml:68]: Unlikely enough that I say let's just hardcode da thing... @monotek thoughts?

Comment (monotek) [reply charts/alertmanager/templates/statefulset.yaml:68]: Imho it does not hurt to have it configurable.

**Review by monotek — APPROVED** [review]

File: charts/alertmanager/templates/statefulset.yaml (hunk 4 above)
Comment (naseemkullah) [reply charts/alertmanager/templates/statefulset.yaml:68]: very well

**Pull request merged**